### PR TITLE
Egg Dropping Puzzle Solution

### LINKED ISSUE #107: Egg Dropping Puzzle
**This issue is up for Python implementation.**

Egg dropping refers to a class of problems in which it is important to find the correct response without exceeding a (low) number of certain failure states. 

**Is your feature request related to a problem? Please describe.**
In a toy example, there is a tower of n floors and an egg dropper with m ideal eggs. The physical properties of the ideal egg are such that it will shatter if it is dropped from floor n^*n or above, and will have no damage whatsoever if it is dropped from floor n^*-1n or below. The problem is to find a strategy such that the egg dropper can determine the floor n^*n in as few eggs drop as possible. 

**Describe the solution you'd like**
This problem has many applications in the real world such as avoiding a call out to the slow HDD or attempting to minimize cache misses or running a large number of expensive queries on a database.

**Additional context**
Content is taken from https://brilliant.org/wiki/egg-dropping/

**Languages**
- [x] Python
- [x] Java
- [ ] C
- [x] C++

### PR DESCRIPTION
**Pull request by codewdhruv**
Added egg drop puzzle solution (Approach: Os)

# Egg Dropping Puzzle Solution

Optimal Substructure: Two possibilities 

- Egg breaks 

- Egg doesnt break

Fixes #107 
`k - Number of floors`
`n - Number of Eggs`
`eggDrop(n, k) - Min no of trials needed to find the worst case scenerio`
`eggDrop(n, k) - 1 + min{max(eggDrop(n – 1, x – 1), eggDrop(n, k – x)), where x is in {1, 2, …, k}}`

### CODE REVIEW

File: C++/eggpuzzle.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/* Contributed by Dhrubajyoti Chakraborty */
+
+/* Egg Drop Puzzle Solution */
+
+#include <bits/stdc++.h>
```
Comment (kaustubhgupta) [inline C++/eggpuzzle.cpp:4]: Please remove extra spaces from the code,

File: C++/eggpuzzle.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            min = res; 
+    } 
+  
+    return min + 1; 
+} 
+  
+ 
+int main() 
+{ 
+    int n = 2, k = 10;
```
Comment (kaustubhgupta) [inline C++/eggpuzzle.cpp]: take user input for this

File: C++/eggpuzzle.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+int main() 
+{ 
+    int n = 2, k = 10; 
+    cout << "Minimum number of trials "
+            "in worst case with "
+         << n << " eggs and " << k 
+         << " floors is "
+         << eggDrop(n, k) << endl; 
+    return 0; 
+}
```
Comment (kaustubhgupta) [inline C++/eggpuzzle.cpp:46]: update the README of this language directory

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: C++/eggpuzzle.cpp (hunk 1 above)
Comment (codewdhruv) [reply C++/eggpuzzle.cpp:4]: fixed

File: C++/eggpuzzle.cpp (hunk 2 above)
Comment (codewdhruv) [reply C++/eggpuzzle.cpp]: fixed

File: C++/eggpuzzle.cpp (hunk 3 above)
Comment (codewdhruv) [reply C++/eggpuzzle.cpp:46]: fixed

**Review by kaustubhgupta — APPROVED** [review]

**Review by YashSaxena9 — APPROVED** [review]

**Comment by YashSaxena9** [comment]
Looks good, thank you for your contributing :)

**Pull request merged**